### ISSUE TITLE
Better test coverage

### ISSUE THREAD

**Issue by bmaupin** [bmaupin/langtrends-data#1]
Do this later, once we have a working POC:

- [x] Add stackoverflow key
- [x] Test with bad keys
- [x] Test GitHub with no key
- ~~Trigger API limits (without a key)~~
  - This seems like a bad idea; it could cause subsequent test failures for myself and for other people

**Comment by bmaupin** [comment]
TL;DR: I think chasing the elusive 100% test coverage is a waste of time and the happy path as well as most common errors seem to be properly handled. We can add more tests on a case-by-case basis as needed.

The remaining uncovered places in the code seem to be related mostly to:

- Code for handling going over API limits
  - While we could test this, we wouldn't want to do it with our API key, or no API key. I guess we could request another one, but this seems like it could potentially violate the terms of service. Seems best left alone
- Other corner cases
  - I guess some of these we could add tests for as they arise, but it would end up being pretty manufactured and I'm not sure how much value it would add. For example, we could remove a `stackoverflowTag` value in languages.json for a particular language to generate a miss, but then the code that handles the normal flow would be missed. Seems messy and low value
